pkg/sdkserver/sdkserver.go

```diff
@@ -824,3 +832,25 @@ func (s *SDKServer) updateConnectedPlayers(ctx context.Context) error {
 	s.recorder.Event(gs, corev1.EventTypeNormal, "PlayerCount", fmt.Sprintf("Set to %d", gs.Status.Players.Count))
 	return nil
 }
+
+// NewSDKServerContext returns a Context that cancels when os.Kill is received
```

Comment:
I don't see a check in the code for os.Kill being received. It looks like in both cases cancel will only be called if the game server state is shutdown. Is that what is intended? 

I read the comment as `( (os.kill) || (os.interrupt && gs.state.shutdown) )` and not `( (os.kill || os.interrupt) && (gs.state.shutdown) )` (the latter of which seems to be what is implemented).

Reply:
Updated the comment to reflect `( (os.kill || os.interrupt) && (gs.state.shutdown) )` which is intended as per the discussions above. If the gs state never reaches shutdown, at the end of the gracefulTermination period, it will get killed regardless, this allows us to clean up the gameserver without having to wait the entire gracefulTermination period